apps/omg_watcher/lib/omg_watcher/block_getter.ex

```diff
@@ -98,15 +99,16 @@ defmodule OMG.Watcher.BlockGetter do
     {:reply, Core.chain_ok(state), state}
   end
 
-  def handle_cast(
-        {:apply_block,
+  def handle_cast({:apply_block, %BlockApplication{} = to_apply}, state), do: apply_block(to_apply, state)
```

Comment:
similar comment as with `LevelDBServer.multi_update`, just here it's simpler (no need for a behavior etc), so maybe let's just invert the approach: have `apply_block` call `GenServer.cast` and decorate that.